cmd/bucket-handlers.go

```diff
@@ -91,8 +92,7 @@ func enforceBucketPolicy(bucket, action, resource, referer, sourceIP string, que
 
 // Check if the action is allowed on the bucket/prefix.
 func isBucketActionAllowed(action, bucket, prefix string, objectAPI ObjectLayer) bool {
-
-	bp, err := objectAPI.GetBucketPolicy(bucket)
+	bp, err := objectAPI.GetBucketPolicy(context.Background(), bucket)
```

Comment:
Not sure whether all the context.Background() are meant as placeholders here and a subsequent PR will fix this. Context needs to be set at handler level and passed all the way down. Otherwise it would seem correct but not serve the purpose of capturing the full context.

Reply:
yes poorna, at a lot of places context.Background() is a placeholder, should be fixed later.